### PR TITLE
feat: add 3d data handling of mesh vertices and faces

### PR DESCRIPTION
**Pull request by anna-charlotte**
Goals:

- Currently, DocArray only supports point clouds for 3d data representation. 
- We want to add functions to additionally support 3d data representation of meshes, which include more information about a 3d object than a point cloud, and seem to be more commonly used. Meshes basically consist of a vertices tensor and a faces tensor, which I plan to load to load to chunks, i.e.:

```python
from docarray import Document

doc = Document(uri='some/uri')
doc.load_uri_to_vertices_and_faces_tensors()

# resulting in:
# doc = Document(
#     uri='some/uri', 
#     chunks=[
#         Document(name='vertices', tensor=sometensor), 
#         Document(name='faces', tensor=sometensor)
#     ]
# )
```

- [x] add `load_uri_to_vertices_and_faces_chunk_tensors`  + tests
- [x] load vertices+faces to point cloud
- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/document/mixins/mesh.py [hunk 1]
```diff
@@ -40,3 +41,57 @@ def load_uri_to_point_cloud_tensor(
...
+            if chunk.tags['name'] == 'faces':
+                faces = chunk.tensor
+
+        if vertices is not None and faces is not None:
+            mesh = trimesh.Trimesh(vertices=vertices, faces=faces)
+            self.tensor = np.array(mesh.sample(samples))
+        else:
+            warnings.warn(
+                'Point cloud tensor can not be set, since vertices and faces chunk tensor have not been set.'
+            )
```
Comment (alaeddine-13) [inline docarray/document/mixins/mesh.py:95]: I think we should raise an error here

**Review by alaeddine-13 — COMMENTED** [review]
we should add documentation here
https://docarray.jina.ai/datatypes/mesh/

File: docarray/document/mixins/mesh.py [hunk 2]
```diff
@@ -40,3 +41,57 @@ def load_uri_to_point_cloud_tensor(
...
+
+        mesh = loader(self.uri, force='mesh')
+
+        vertices = mesh.vertices.view(np.ndarray)
+        faces = mesh.faces.view(np.ndarray)
+
+        self.chunks = [
+            Document(name='vertices', tensor=vertices),
+            Document(name='faces', tensor=faces),
+        ]
```
Comment (alaeddine-13) [inline docarray/document/mixins/mesh.py:66]: maybe let's enforce the schema here with `id` and later retrieve with `self.chunks['vertices']`

File: docarray/document/mixins/mesh.py [hunk 3]
```diff
@@ -40,3 +41,57 @@ def load_uri_to_point_cloud_tensor(
             self.tensor = np.array(mesh.sample(samples))
 
         return self
+
+    def load_uri_to_vertices_and_faces_chunk_tensors(self: 'T') -> 'T':
```
Comment (samsja) [inline docarray/document/mixins/mesh.py:45]:
```suggestion
    def load_uri_to_vertices_and_faces(self: 'T') -> 'T':
```

name is too long IMO (even though it stick to our old naming convention)

File: docarray/document/mixins/mesh.py [hunk 4]
```diff
@@ -40,3 +41,57 @@ def load_uri_to_point_cloud_tensor(
...
+        faces = mesh.faces.view(np.ndarray)
+
+        self.chunks = [
+            Document(name='vertices', tensor=vertices),
+            Document(name='faces', tensor=faces),
+        ]
+
+        return self
+
+    def load_vertices_and_faces_chunk_tensors_to_point_cloud_tensor(
```
Comment (samsja) [inline docarray/document/mixins/mesh.py:70]: same lets rename here

File: docarray/document/mixins/mesh.py (hunk 2 above)
Comment (alaeddine-13) [reply docarray/document/mixins/mesh.py:66]: alright makes sense then

File: docs/datatypes/mesh/index.md [hunk 5]
```diff
@@ -31,3 +32,36 @@ The following pictures depict a 3D mesh and a point cloud with 1000 samples from
 ```{figure} pointcloud-man.gif
 :width: 50%
 ```
+
+## Mesh
```
Comment (alaeddine-13) [inline docs/datatypes/mesh/index.md:36]: the section title is `3D Mesh` above, and then we explain point cloud and go back to mesh.
It feels a bit weird to me, maybe we should name this one `Vertices and Faces representation` and the other one `point cloud representation` ?

Comment (alaeddine-13) [reply docs/datatypes/mesh/index.md]: yes I agree

**Review by samsja — APPROVED** [review]
lgtm to me as well. @JohannesMessner what do u think about the naming of the function ?

File: docs/datatypes/mesh/index.md [hunk 6]
```diff
@@ -7,27 +7,63 @@ This feature requires `trimesh`. You can install it via `pip install "docarray[f
...
 ```
 
+## Point cloud representation
+
+Point cloud is a representation of a 3D mesh. It is made by repeated and uniformly sampling points within the 3D body. Comparing to the mesh representation, point cloud is a fixed size ndarray and hence easier for deep learning algorithms to handle. In DocArray, you can simply load a 3D mesh and convert it into a point cloud via:
+
+```python
+from docarray import Document
+
+doc = Document(uri='viking.glb').load_uri_to_point_cloud_tensor(1000)
```
Comment (JohannesMessner) [inline docs/datatypes/mesh/index.md:56]: let's explain this `1000` (one sentence is enough), even if it is maybe clear from the context.
Also, if it has a keyword we should pass it using this keyword, for increased clarity.

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

File: docs/datatypes/mesh/index.md [hunk 7]
```diff
@@ -7,27 +7,63 @@ This feature requires `trimesh`. You can install it via `pip install "docarray[f
 
 A 3D mesh is the structural build of a 3D model consisting of polygons. Most 3D meshes are created via professional software packages, such as commercial suites like Unity, or the free open source Blender 3D.
 
-## Point cloud
+## Vertices and faces representation 
 
-Point cloud is a representation of a 3D mesh. It is made by repeated and uniformly sampling points within the 3D body. Comparing to the mesh representation, point cloud is a fixed size ndarray and hence easier for deep learning algorithms to handle. In DocArray, you can simply load a 3D mesh and convert it into a point cloud via:
+A 3D mesh can be represented by its vertices and faces. Vertices are points in a 3D space, represented as a tensor of shape (n_points, 3). Faces are triangular surfaces that can be defined by three points in 3D space, corresponding to the three vertices of a triangle. Faces can be represented as a tensor of shape (n_faces, 3). Each number in that tensor refers to an index of a vertex in the tensor of vertices.
+
+In DocArray, you can load a mesh and save its vertices and faces to a Documents `.chunks` as follows:
```
Comment (NicholasDunham) [inline docs/datatypes/mesh/index.md:14]:
```suggestion
In DocArray, you can load a mesh and save its vertices and faces to a Document's `.chunks` as follows:
```

File: docs/datatypes/mesh/index.md [hunk 8]
```diff
@@ -7,27 +7,63 @@ This feature requires `trimesh`. You can install it via `pip install "docarray[f
...
 ```text
-(1000, 3)
+ <Document ('id', 'chunks') at 7f907d786d6c11ec840a1e008a366d49>
+    └─ chunks
+          ├─ <Document ('id', 'parent_id', 'granularity', 'tensor', 'tags') at 7f907ab26d6c11ec840a1e008a366d49>
+          └─ <Document ('id', 'parent_id', 'granularity', 'tensor', 'tags') at 7f907c106d6c11ec840a1e008a366d49>
 ```
 
-The following pictures depict a 3D mesh and a point cloud with 1000 samples from that 3D mesh. 
+This stores the vertices and faces in `.tensor` of two separate sub-Documents in a Documents `.chunks`. Both sub-Documents have a name assigned to them ('vertices' or 'faces''), which is saved in `.tags`:
```
Comment (NicholasDunham) [inline docs/datatypes/mesh/index.md:31]:
```suggestion
This stores the vertices and faces in `.tensor` of two separate sub-Documents in a Document's `.chunks`. Each sub-Document has a name assigned to it ('vertices' or 'faces'), which is saved in `.tags`:
```

File: docs/datatypes/mesh/index.md [hunk 9]
```diff
@@ -7,27 +7,63 @@ This feature requires `trimesh`. You can install it via `pip install "docarray[f
...
+
+The following picture depicts a 3D mesh:
 
 ```{figure} 3dmesh-man.gif
 :width: 50%
 ```
 
+## Point cloud representation
+
+Point cloud is a representation of a 3D mesh. It is made by repeated and uniformly sampling points within the 3D body. Comparing to the mesh representation, point cloud is a fixed size ndarray and hence easier for deep learning algorithms to handle. In DocArray, you can simply load a 3D mesh and convert it into a point cloud via:
```
Comment (NicholasDunham) [inline docs/datatypes/mesh/index.md:51]:
```suggestion
A point cloud is a representation of a 3D mesh. It is made by repeatedly and uniformly sampling points within the 3D body. Compared to the mesh representation, the point cloud is a fixed size ndarray and hence easier for deep learning algorithms to handle. In DocArray, you can simply load a 3D mesh and convert it into a point cloud via:
```

**Review by NicholasDunham — CHANGES_REQUESTED** [review]

**Review by hanxiao — APPROVED** [review]

**Pull request merged**